src/lib/universal-client.js

```diff
@@ -0,0 +1,579 @@
+/* cometvisu.js (c) 2010 by Christian Mayer [CometVisu at ChristianMayer dot de]
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
+ */
+
+/**
+ * The JavaScript library that implements the CometVisu protocol.
+ * @module CometVisu Client
+ * @title  CometVisu Client
+ * @reqires jQuery
+*/
+define( [], function() {
+  "use strict";
+  
+/**
+ * Class that handles the communicaton of the client * 
+ * @class CometVisu
+ * @constructor foo
+ * @param backend
+ */
+function CometVisu( backend, initPath )
+{
+  this.initPath = initPath ? initPath : null;
+  
+  //used for backwards compability
+  this.aliases = {
+      'cgi-bin' : 'default',
+      'oh'      : 'openhab',
+      'oh2'     : 'openhab2'
+  };
+  if (this.aliases[backend]) {
+    backend = this.aliases[backend];
+  }
+  
+  this.backends = {
+	  'default' : {
+  		name 	  : 'default',
+  		baseURL : '/cgi-bin/',
+  		transport : 'long-polling',
+  		resources : {
+  		  login : 'l',
+  		  read 	: 'r',
+  		  write : 'w',
+  		  rrd 	: 'rrdfetch'
+  		},
+  		hooks : {}
+	  },
+	  'openhab' : {
+  		name 		: 'openHAB',
+  		baseURL 	: '/services/cv/',
+  		// keep the e.g. atmosphere tracking-id if there is one
+  		resendHeaders : {
+  		  'X-Atmosphere-tracking-id' : null
+  		},
+  		// fixed headers that are send everytime
+  		headers : {
+  		  'X-Atmosphere-Transport' : 'long-polling'
+  		},
+  		hooks : {
+  		  onClose : function() {
+  		    // send an close request to the openHAB server
```

Comment:
Please replace Tabs by spaces - otherwise the code alignment might break as demonstrated on this line above